### PR TITLE
dist: add commit message length checker

### PR DESCRIPTION
**Pull request by miri64**
This checks if the commit messages of a PR follow the [50/72 rule](https://medium.com/@preslavrachev/what-s-with-the-50-72-rule-8a906f61f09c). This way we don't merge these kind of commit messages accidentally anymore ;-).

### CODE REVIEW

**Comment by miri64** [comment]
Why is Travis not executing that script?

**Comment by miri64** [comment]
Mh... if I run `make static-tests` locally, it works...

**Comment by miri64** [comment]
> Why is Travis not executing that script?

(I was mistaken with that assessment... I'm just confused, why it doesn't work, while it works locally for me)

**Comment by miri64** [comment]
Ahh... could it be, that the empty commits are removed by this?

**Comment by miri64** [comment]
Works now with a non-empty commit! :-)

**Comment by aabadie** [comment]
How does it work with multiline commit messages ? Maybe we want to keep those ones ?

**Comment by miri64** [comment]
> How does it work with multiline commit messages ? Maybe we want to keep those ones ?

As long as contributors stick to the rules for those (i.e. start the detailed description after one empty line after the summary) the check will only complain about the summary (`--pretty=format:"%s"` makes sure of that). There are already commits in the history who ignore this rule. Naturally, Git then thinks the detailed description belongs to the summary and will complain about this in this check.

**Comment by miri64** [comment]
As you can see in https://travis-ci.org/RIOT-OS/RIOT/builds/311342120: 79cfa95 doesn't show up, but d268896 still does.

File: dist/tools/commit-msg/check.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+# Copyright 2015 Oliver Hahm <oliver.hahm@inria.fr>
```
Comment (aabadie) [inline dist/tools/commit-msg/check.sh]: Is this the right copyright ?

**Comment by gebart** [comment]
I like :+1:

**Comment by miri64** [comment]
Ping?

**Comment by miri64** [comment]
Done. Did a minor indentation fix to the script ;-)

**Comment by aabadie** [comment]
All green, go!

**Comment by kaspar030** [comment]
I'm not sure I like this... *all* my commit descriptions are longer than 50 characters... ;)

**Comment by miri64** [comment]
> I'm not sure I like this... all my commit descriptions are longer than 50 characters... ;)

Well it is BCP for git, so you better should adapt your git behavior ;-). Also: those only get a warning, the 72 ones are the ones that cause errors ;-).

**Pull request merged**